### PR TITLE
DDT-1100: EnsurePhaseTypeCodeIsDefined refactor

### PR DESCRIPTION
**Pull request by mp-assco**

### CODE REVIEW

File: jit-integration-tests/src/main/java/org/dcsa/jit/itests/v1/PostTimestampsIT.java [hunk 1]
```diff
@@ -503,42 +503,6 @@ public void testMandatoryUNLocationCodeFieldFalseFormat() {
...
-    // Empty value
-    map.put("facilityTypeCode", ""); // Empty
-    given()
-        .contentType("application/json")
-        .body(map)
-        .post(TIMESTAMPS)
-        .then()
-        .assertThat()
-        .statusCode(400);
-  }
```
Comment (nt-as) [inline jit-integration-tests/src/main/java/org/dcsa/jit/itests/v1/PostTimestampsIT.java:540]: I think we should keep this test.  The first part might need corrections to still succeed (as there are definitely timestamps where `facilityTypeCode` is mandatory).

File: jit-service/src/main/java/org/dcsa/jit/service/TimestampDefinitionService.java [hunk 2]
```diff
@@ -61,6 +62,69 @@ public void markOperationsEventAsTimestamp(OperationsEvent operationsEvent) {
...
+            "More than one JIT 1.1 timestamp definitions found according to the given fields: "
+                + errorMessage);
+      }
+
+      if (jit1_1.isEmpty()) {
+        throw ConcreteRequestErrorMessageException.internalServerError(
+            "No JIT 1.1 timestamp definitions found according to the given fields: "
+                + errorMessage);
+      }
+      return jit1_1.stream().findFirst().get().getPortCallPhaseTypeCode();
```
Comment (nt-as) [inline jit-service/src/main/java/org/dcsa/jit/service/TimestampDefinitionService.java:116]: I think we should be strict and reject JIT 1.1+ timestamps without port call phase type code (unless the timestamp explicitly permit it to be null - none do right now as I recall, but some will soon).

File: jit-integration-tests/src/main/java/org/dcsa/jit/itests/v1/PostTimestampsIT.java (hunk 1 above)
Comment (mp-assco) [reply jit-integration-tests/src/main/java/org/dcsa/jit/itests/v1/PostTimestampsIT.java]: Good point, test case restored.

File: jit-service/src/main/java/org/dcsa/jit/service/TimestampDefinitionService.java [hunk 3]
```diff
@@ -125,6 +101,27 @@ public PortCallPhaseTypeCode findPhaseTypeCodeFromOperationsEventForJit1_0(
     return jit1_0.stream().findFirst().get().getPortCallPhaseTypeCode();
   }
 
+  public PortCallPhaseTypeCode findOmittedPhaseTypeCodeFromOperationsEventForJit1_1(
+      OperationsEvent operationsEvent) {
+
+    // JIT 1.1 use-cases with omitted port call phase type code
+    List<String> phaseTypeCodeOmitted = IntStream.range(5, 13).boxed().map(i -> "UC" + i).toList();
```
Comment (nt-as) [inline jit-service/src/main/java/org/dcsa/jit/service/TimestampDefinitionService.java:108]: These timestamps should be corrected in the IM til be from `jit1_0` instead.   I will do that  in my next IM update

**Review by nt-as — APPROVED** [review]

**Pull request merged**